### PR TITLE
Add scripts to extract workflows from WorkflowHub and UseGalaxy servers

### PR DESCRIPTION
**Pull request by bebatut**

### CODE REVIEW

**Comment by nsoranzo** [comment]
@bebatut I can help with the linting on Python 3.8, if you like.

**Comment by bebatut** [comment]
> @bebatut I can help with the linting on Python 3.8, if you like.

Thanks. I wanted to know if we wanted to keep it as all other tests are made with Python 3.11. Do you think we should still keep the linting for Python 3.8?

I also have no idea how to fix the other tests so that they get the PR branch but also the content of the result branch. Any idea?

**Comment by bebatut** [comment]
> Fixed the linting, don't have time to look at the rest at the moment, sorry!

Thanks already for fixing the linting!!

File: sources/bin/shared.py [hunk 1]
```diff
@@ -82,3 +82,29 @@ def get_request_json(url: str, headers: dict) -> dict:
...
 def format_date(date: str) -> str:
     return datetime.fromisoformat(date).strftime("%Y-%m-%d")
+
+
+def shorten_tool_id(tool: str) -> str:
+    """
+    Shorten tool id
+    """
+    if "toolshed" in tool:
+        return tool.split("/")[-2]
```
Comment (paulzierep) [inline sources/bin/shared.py:92]: are you sure it's not `[-1]` ?

**Review by paulzierep — COMMENTED** [review]
Could you add the output of a test run ? That would make the review simpler. Will try to check tomorrow.

File: sources/bin/shared.py (hunk 1 above)
Comment (bebatut) [reply sources/bin/shared.py:92]: If we want to have the suite id it is `[-2]` and it is what we used there: https://github.com/galaxyproject/galaxy_codex/pull/158/files#diff-fded0df90a63fa85d73a79a61797665ed61e1b1a921558f0d3b301d877a54dc4L38

**Review by paulzierep — APPROVED** [review]
Looks good and I like the OO approach, could you provide the test output, still ?

**Comment by paulzierep** [comment]
workflow CI fails, with
```
        wfhub_wfs = shared.get_request_json(
            "https://workflowhub.eu/workflows?filter[workflow_type]=galaxy",
            header,
        )
```
locally it worked for me, maybe we need to retry this a few times ... ?

**Review by paulzierep — APPROVED** [review]
Really nice ! Thanks @bebatut !

**Pull request merged**